zipline/assets/assets.py

```diff
@@ -440,22 +440,27 @@ def _select_assets_by_sid(asset_tbl, sids):
     def _select_asset_by_symbol(asset_tbl, symbol):
         return sa.select([asset_tbl]).where(asset_tbl.c.symbol == symbol)
 
-    def _lookup_most_recent_symbols(self, sids):
+    def _select_most_recent_symbols_chunk(self, sid_group):
         symbol_cols = self.equity_symbol_mappings.c
+        inner = sa.select(
+            (symbol_cols.sid,) +
```

Comment:
How often is this function called?  Would it be worthwhile to cache the first part of this select, since it doesn't depend on the input?